test/quantization/test_quant_primitives.py

```diff
@@ -838,6 +838,32 @@ def test_fake_quantize_affine_cachemask(self):
         torch.testing.assert_close(dequantized, fake_quantized)
         torch.testing.assert_close(expected_mask, mask)
 
+    # ZeroPointDomain.NONE should work
+    def test_none_zero_point_domain(self):
```

Comment:
we could also have a test for zero_point_domain being None and throw an IllegalArgumentError exception I think

Reply:
Thanks again for reviewing! I added code for raising `ValueError` if `zero_point_domain` would be `None`, but only for `choose_qparams_affine` and `choose_qparams_affine_with_min_max`.

For other public-facing APIs that have `zero_point_domain` as a parameter, I added asserts instead.

Please advise if this is fine, or if other places in the code should also throw exceptions instead of asserting.

Thanks!